### PR TITLE
chore(fe): Fix deprecations and breaking changes in deps

### PR DESCRIPTION
**Pull request by lhvy**

### CODE REVIEW

File: frontend/src/pages/TermPlanner/TermBox/styles.ts [hunk 1]
```diff
@@ -32,25 +32,25 @@ const TermBoxWrapper = styled.ul<TermBoxWrapperProps>`
...
+    $summerEnabled &&
     css`
       min-height: 13.5em;
       min-width: 13em;
     `}
 `;
 
-const TermCheckboxWrapper = styled.div<{ checked: boolean }>`
-  background-color: ${({ checked }) => (checked ? '#9685f3' : '#D9D9D9')};
+const TermCheckboxWrapper = styled.div<{ $checked: boolean }>`
```
Comment (ollibowers) [inline frontend/src/pages/TermPlanner/TermBox/styles.ts:52]: `frontend/src/pages/TermPlanner/TermBox/TermBox.tsx:107` is not converted to transient 😡

you forgor 💀

**Review by ollibowers — CHANGES_REQUESTED** [review]
other then that one missed spot, looks good to me

File: frontend/src/pages/TermPlanner/TermBox/styles.ts (hunk 1 above)
Comment (ollibowers) [reply frontend/src/pages/TermPlanner/TermBox/styles.ts:52]: dont forget to run `npm run types`, silly

Comment (lhvy) [reply frontend/src/pages/TermPlanner/TermBox/styles.ts:52]: Oops. Fixed thanks.

**Review by ollibowers — APPROVED** [review]
lgtm

**Pull request merged**